src/main/java/com/fasterxml/jackson/databind/util/StdDateFormat.java

```diff
@@ -131,12 +124,9 @@
      */
     protected Boolean _lenient;
     
-    protected transient DateFormat _formatRFC1123;
-    protected transient DateFormat _formatISO8601;
-    protected transient DateFormat _formatISO8601_z;
-    protected transient DateFormat _formatISO8601_noTz; // 2.8.10
-    protected transient DateFormat _formatPlain;
 
+    private transient Map<DateFormat, SimpleDateFormat> _clonedFormats = new HashMap<>();
```

Comment:
Enum is a good option indeed - I think we could go that way.
ThreadLocal is another I thought off. We could even build something like a DateFormatCache/factory from which we would ask for pre-configured instances. This cache could be used at different places: by the StdDateFormat but maybe also by the DateBasedDeserializer for when it creates Contextual instances. I would love to add something like the baseline vs. local DateFormat instances like what you did here initially. I planned to look at that possibility just after...